### ISSUE TITLE
CSine<> Amplitude drift

### ISSUE THREAD

**Issue by mhetrick** [LancePutnam/Gamma#44]
We're having issues with CSine<> changing amplitude very slowly over time, even with decay = -1. We're experiencing audible amplitude drift after 30+ minutes.

**Comment by mhetrick** [comment]
As a workaround, it appears that the amplitude drift doesn't occur when using CSine<double>.

**Comment by LancePutnam** [comment]
Yes, for very long decays, double precision is required. For no decay, since single precision may be faster (it is on all my machines), another workaround may be to normalize 'val' every so often. Since I can't think of an automatic way to prevent drift that doesn't incur extra overhead (on each sample), I've added a note to the CSine comment section.

**Comment by mhetrick** [comment]
We have no decay on ours. It runs away after 30 minutes with single precision. It turns out that double precision didn't fix it either, as it runs away after a longer period of time (even with no decay).

We'll look into automatic normalization and see if we can fix it that way without too much overhead.
